zebra/zebra_rib.c

```diff
@@ -1937,7 +1937,7 @@ static int rib_count_installed_nh(struct route_entry *re)
 		/* The meaningful flag depends on where the installed
 		 * nexthops reside.
 		 */
-		if (nhg == &(re->fib_backup_ng)) {
+		if (nhg == &(re->fib_ng)) {
```

Comment:
We are changing this from backup -> primary?

What is this check actually for?

Reply:
I think this was a typo from ... a while ago. what's happening here is counting installed nexthops. the rule is that in the 'all or nothing' case, every rib nexthop is considered installed if it's marked ACTIVE. but if there is a fib-specific set, then those are individually flagged with the FIB flag.

the backups are a bit different - the only installed backups are in the fib-specific set, and that api _can_ return a NULL, as the SA admonished me.